src/python/gudhi/cover_complex.py

```diff
@@ -17,6 +17,10 @@
 from sklearn.metrics         import pairwise_distances
 from scipy.spatial.distance  import directed_hausdorff
 
+from sklearn import __version__ as sklearn_version
+from pkg_resources import packaging
```

Comment:
Wait, does that come with Python, or is it a new dependency? It seems deprecated...

Reply:
@mglisse It seems to be the way it is done by setuptools (and it is required by our python package), cf. https://peps.python.org/pep-0386/#setuptools (I have read it is available for python 3.6+ in this [this stackoverflow](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python), but I cannot find the information)

Reply:
https://setuptools.pypa.io/en/latest/pkg_resources.html says
> Use of pkg_resources is deprecated in favor of [importlib.resources](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata) and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [packaging](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of pkg_resources and should work to port to importlib-based solutions.

setuptools is currently a dependency for **building** gudhi, but not for **running** it, so using it there would require some configuration changes.